### ISSUE TITLE
2.4 version give an extra output when running inference in CPU with a Huggingface model.. traced to a dynamo bug

### ISSUE THREAD

**Issue by mserranos** [pytorch/pytorch#135353]
### 🐛 Describe the bug

When running Roberta Question Answering (and also other Huggingface models) in CPU inference mode, I get an extra output returned by dynamo, that did not happen in a previous 2.3.1 version (see yellow extra output):
![image](https://github.com/user-attachments/assets/d3ea3750-7ca1-44a0-ad0f-7017d0cb116f)

This is a bug in _dynamo/symbolic_convert.py, which seems to be picking out an extra item "hidden_states_95" item, even though we know that output.hidden_states is Null:

[bug.txt](https://github.com/user-attachments/files/16908602/bug.txt)

### Versions

2.4.0, 2.4.1

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @amjames @rec

**Comment by anijain2305** [comment]
@mserranos Can you please provide the repro instructions?

**Comment by anijain2305** [comment]
I was able to repro this. A couple of observations

```
python benchmarks/dynamo/huggingface.py --performance --cold-start-latency --training --amp --backend eager --device cpu --only=RobertaForQuestionAnswering
```


1) This is **NOT** an accuracy problem. Yes, there is one more graph output in the Dynamo Fx graph. But Dynamo also generates post graph bytecode, and that stitches the data structures correctly to ignore this extra output. I checked by putting breakpoint after the model is called, and comparing the `main` branch and `2.4.0`. The return of the model still has `hidden_output` attribute as None.

2) This extra output is fixed on the main branch. The fix landed here - https://github.com/pytorch/pytorch/pull/127780

Removing high-priority as this is not an accuracy regression. 

@mserranos If there is a hard dependency for you to not have hidden_states as output, then moving to pytorch-nightly is the best option. `torch.compile` does not guarantee Dynamo Fx graph to be same across versions.

**Comment by StrongerXi** [comment]
@anijain2305 looks like we can close this?

**Comment by anijain2305** [comment]
Closing this as this is not a problem.
